src/services/firebase/notifications.ts

```diff
@@ -0,0 +1,245 @@
+        title: `${currencyName} sent`,
+        body: `Safe ${shortenAddress(address)} on ${chainName} sent ${formatUnits(
+          value,
+          chain?.nativeCurrency?.decimals,
+        ).toString()} ${currencySymbol} in transaction ${shortenAddress(txHash)}.`,
+      }
+    },
+    [WebhookType.INCOMING_TOKEN]: async ({ address, txHash, tokenAddress, value }) => {
+      const token = await getTokenInfo(data.chainId, address, tokenAddress, value)
+      return {
+        title: `${token.name} received`,
+        body: `Safe ${shortenAddress(address)} on ${chainName} received ${token.value} ${
+          token.symbol
+        } in transaction ${shortenAddress(txHash)}.`,
+      }
+    },
+    [WebhookType.OUTGOING_TOKEN]: async ({ address, txHash, tokenAddress, value }) => {
+      const token = await getTokenInfo(data.chainId, address, tokenAddress, value)
+      return {
+        title: `${token.name} sent`,
+        body: `Safe ${shortenAddress(address)} on ${chainName} sent ${token.value} ${
+          token.symbol
+        } in transaction ${shortenAddress(txHash)}.`,
+      }
+    },
+    [WebhookType.MODULE_TRANSACTION]: ({ address, module, txHash }) => {
+      return {
+        title: 'Module transaction',
+        body: `Safe ${shortenAddress(address)} on ${chainName} executed a module transaction ${shortenAddress(
+          txHash,
+        )} from module ${shortenAddress(module)}.`,
+      }
+    },
+    [WebhookType.CONFIRMATION_REQUEST]: ({ address, safeTxHash }) => {
+      return {
+        title: 'Confirmation request',
+        body: `Safe ${shortenAddress(
+          address,
+        )} on ${chainName} has a new confirmation request for transaction ${shortenAddress(safeTxHash)}.`,
+      }
+    },
+    [WebhookType.SAFE_CREATED]: () => {
+      // We do not preemptively subscribe to Safes before they are created
+      return null
+    },
+  }
+
+  // Can be safely casted as `data.type` is a mapped type of `NotificationsMap`
+  const notification = await Notifications[data.type](data as any)
+
+  if (notification) {
+    return {
+      ...notification,
+      link: getLink(data, chain?.shortName),
+    }
+  }
+}
+
+export const parseFirebaseNotification = async (
+  payload: MessagePayload,
+): Promise<({ title: string; link?: string } & NotificationOptions) | undefined> => {
+  // Transaction Service-dispatched notification
+  if (isWebhookEvent(payload.data)) {
+    const webhookNotification = await _parseWebhookNotification(payload.data)
+
+    if (webhookNotification) {
+      return {
+        title: webhookNotification.title,
+        body: webhookNotification.body,
+        link: webhookNotification.link,
+      }
+    }
+  }
+
+  // Manually dispatched notifications from the Firebase admin panel
+  // Displayed as is
+  if (payload.notification) {
+    return {
+      title: payload.notification.title || '',
+      body: payload.notification.body,
+      image: payload.notification.image,
+    }
+  }
```

Comment:
I feel like this could be condensed into two-three lines?

Reply:
Addressed in bbe73f4277ad22d090b588b66f45f57e3d06d560.